File: src/communityday/islands.js
// Javascript sorting by a property is a lot of work

var confs = [{name : "Clojure Conj", year : 2011 },
	     {name : "Community Day", year : 2012 }];

var sorted = confs.sort(function(a, b) {
 if (a.name < b.name) //sort string ascending
  return -1 
 if (a.name > b.name)
  return 1
 return 0 //default return value (no sorting)
});

print("Confs:");
print(JSON.stringify(confs));
print("Sorted:");
print(JSON.stringify(sorted));

File: src/communityday/islands.clj
(ns communityday.islands)

;; A Clojure record is not an island
(defrecord Conference [name year])

(def cc (Conference. "Clojure Conj" 2011))
(def cday (Conference. "Community Day" 2012))
(def confs [cc cday])

;; Records works with common functions
(filter #(= 2011 (:year %)) confs)
(sort-by :name confs)

;; a String is a Seq, and Seqs can be counted
;; Sort by the length of the name:
(sort-by (fn [c] (count (:name c))) confs)

;; Fields can be added dynamically
(assoc cday :rating :great)

;; Conference fields have map semantics
(:year cday)

;; A record is also a map of its properties
(seq cday)

(for [[property value] cday]
  (str property "->" value))

